qt/widgets/common/src/AlgorithmHistoryWindow.cpp

```diff
@@ -696,6 +736,28 @@ void AlgHistoryTreeWidget::uncheckAllChildren(QTreeWidgetItem *item,
   }
 }
 
+void AlgHistoryTreeWidget::removeHiddenChildren(QTreeWidgetItem *item) {
+  if (auto parent = item->parent()) {
+    int index{parent->indexOfChild(item)};
+
+    for (int i = 0; i < item->childCount(); i++) {
+      if (item->child(i)->checkState(1) == Qt::Checked) {
+        removeHiddenChildren(item->child(i));
+      }
+      delete parent->child(index + 1);
```

Comment:
I find the logic difficult to follow here. It looks like it's deleting memory but not removing it from the child list? And it always deletes at the same location?

Reply:
I'm back on this PR now. The QT docs say that using 'delete' is a safe way to remove an item from a tree widget:
![image](https://user-images.githubusercontent.com/52415735/72601409-4698d280-390d-11ea-867c-7722248c2b4c.png)

The reason it's always at the same location is because when you remove an item, all of the subsequent items "move up" in the list if that makes sense. So each item being deleted ends up at the same index.